File: problem_010.swift
#!/usr/bin/xcrun swift
// -*- compile-command: "swiftc problem_010.swift -o problem_010" -*-
// Copyright (c) 2016 Michael Caldwell
import Cocoa

print("Project Euler - Problem 10:")
print("Find the sum of all the primes below two million.\n")

func isPrime(n:Int) -> Bool {
    if n < 2 { return false }
    if n == 2 { return true }
    if n % 2 == 0 { return false }

    var i = 3
    while i * i <= n {
        if n % i == 0 {
            return false
        }
        i += 2
    }

    return true;
}

let limit = 2000000
var sum = 2

for var n in stride(from: 3, to: limit, by: 2) {
    if isPrime(n: n) == true {
        sum += n
    }
}

print(sum)

File: problem_010.go
// -*- compile-command: "go build -o problem_010_go problem_010.go" -*-
// Copyright (c) 2017 Michael Caldwell
package main

import "fmt"

func main() {
        fmt.Println("Project Euler - Problem 10:")
        fmt.Println("Find the sum of all the primes below two million.\n")

	var sieve [2000000]bool
	limit := len(sieve)
	sum := 0

	for n:=2; n<limit; n++ {
		if sieve[n] == false {
			sum += n
			for i := n; i<limit; i += n {
				sieve[i] = true
			}
		}
	}

        fmt.Println(sum)
}
